<?xml version="1.0"?>
<launch>
  <param name="robot_description" command="$(find xacro)/xacro.py '$(find sick_tim)/urdf/example.urdf.xacro'" />
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />
  <node name="sick_tim571_2050101" pkg="sick_tim" type="sick_tim551_2050001" respawn="false" output="screen">
    <!-- default values: -->
    
      <param name="min_ang" type="double" value="-2.35619449019" />
      <param name="max_ang" type="double" value="2.35619449019" />
      <param name="intensity" type="bool" value="True" />
      <param name="skip" type="int" value="0" />
      <param name="frame_id" type="str" value="laser" />
      <param name="time_offset" type="double" value="-0.001" />
      <param name="publish_datagram" type="bool" value="False" />
      <param name="subscribe_datagram" type="bool" value="false" />
      <param name="time_increment" type="double" value="0.000061722" />
      <param name="range_min" type="double" value="0.05" />
      <param name="range_max" type="double" value="30.0" />
    
<!-- Enables TCP instead of USB connection; `hostname` is the host name or IP address of the laser scanner. 
Change the IP address according to your network and device Ip address. 
    In cases where a race condition exists and the computer boots up before the TIM is ready, increase 'timelimit.'-->

         <param name="hostname" type="string" value="192.168.1.21" />
         <param name="port" type="string" value="2112" />
         <param name="timelimit" type="int" value="5" />
    
  </node>
</launch>
